### PR TITLE
GisticReader bugs, fixes Issue #884

### PR DESCRIPTION
**Pull request by zheins**
Split genes by | and take only front portion, and if no genes found in database we should skip the event instead of adding invalid entry to database.

### CODE REVIEW

**Comment by zheins** [comment]
#884

File: core/src/main/java/org/mskcc/cbio/portal/util/GisticReader.java [hunk 1]
```diff
@@ -200,25 +204,26 @@ private Gistic parseLine(String line, int cancerStudyId, int chromosomeField, in
...
 
-            if (canonicalGene == null) {
-                canonicalGene = new CanonicalGene(gene);
-
-//                    System.out.println("gene not found, skipping: " + gene);
-//                    throw new DaoException("gene not found: " + gene);
-            }
+            if (canonicalGene != null) {
+                if (canonicalGene.isMicroRNA()) {
+                    System.err.println("ignoring miRNA: " + canonicalGene.getHugoGeneSymbolAllCaps());
```
Comment (pieterlukasse) [inline core/src/main/java/org/mskcc/cbio/portal/util/GisticReader.java]: Please use
`ProgressMonitor.logWarning()` method. In this way all warnings are grouped together and reported at the end of the import step

File: core/src/main/java/org/mskcc/cbio/portal/util/GisticReader.java [hunk 2]
```diff
@@ -200,25 +204,26 @@ private Gistic parseLine(String line, int cancerStudyId, int chromosomeField, in
...
+                genes.add(canonicalGene);
             }
-
-            genes.add(canonicalGene);
         }
         // -- end parse genes --
-
+        
+        if (genes.size() == 0) {
+            logger.info("No genes found in database - skipping gistic event");
```
Comment (pieterlukasse) [inline core/src/main/java/org/mskcc/cbio/portal/util/GisticReader.java]: Please use
`ProgressMonitor.logWarning()` method. In this way all warnings are grouped together and reported at the end of the import step

**Comment by pieterlukasse** [comment]
@zheins : thanks for the fix. I added some minor comments regarding the logging.

File: core/src/main/java/org/mskcc/cbio/portal/util/GisticReader.java (hunk 1 above)
Comment (pieterlukasse) [inline core/src/main/java/org/mskcc/cbio/portal/util/GisticReader.java]: I will do it.

File: core/src/main/java/org/mskcc/cbio/portal/util/GisticReader.java (hunk 2 above)
Comment (pieterlukasse) [inline core/src/main/java/org/mskcc/cbio/portal/util/GisticReader.java]: I will do it.

**Pull request merged**